### PR TITLE
[WIP]  Add a way to register resources with the rest root

### PR DESCRIPTION
**Pull request by wizhippo**
This adds the ability to add resources in the rest root so rest consumers can discover them.  For example you can register the resource and use the capi auto discovery to find the uri for the resource.

To add a resource a bundle would add the resource using the PrependExtensionInterface and pre-pending the new resource to the config.

### CODE REVIEW

**Comment by andrerom** [comment]
Why did you close this already?

**Comment by wizhippo** [comment]
@andrerom If there is interest I will re-open.

**Comment by andrerom** [comment]
@bdunogier POV?

**Comment by emodric** [comment]
I'm interested :)

It would be convenient to be able to add path to load a tag over there for example.

**Comment by wizhippo** [comment]
I have reopen at https://github.com/ezsystems/ezpublish-kernel/pull/1619 as I had rebased it on master

File: eZ/Publish/Core/REST/Common/Values/Resource.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+namespace eZ\Publish\Core\REST\Common\Values;
+
+use eZ\Publish\Core\REST\Common\Value as RestValue;
+
+/**
+ * This class represents a resource.
+ */
+class Resource extends RestValue
+{
+    public $name;
```
Comment (bdunogier) [inline eZ/Publish/Core/REST/Common/Values/Resource.php]: Name and example on all public properties, especially on a value object. Please :-)

File: eZ/Publish/Core/REST/Common/Values/Root.php [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * File containing the Root class.
+ * File containing the Resource class.
```
Comment (bdunogier) [inline eZ/Publish/Core/REST/Common/Values/Root.php]: Hmmm, maybe not :-)

Comment (andrerom) [inline eZ/Publish/Core/REST/Common/Values/Root.php]: @wizhippo do you take these comments into account on the new PR?

Comment (wizhippo) [inline eZ/Publish/Core/REST/Common/Values/Root.php]: Yes, I just have just pushed them,

**Pull request closed without merging**